packages/wrangler/src/versions/utils/config.ts

```diff
@@ -1,13 +1,11 @@
-import path from "path";
+import path from "node:path";
 import { findWranglerConfig, readConfig } from "../../config";
 
-export function getConfig<
-	T extends {
-		name?: string;
-		config?: string;
-	},
->(args: Pick<T, "config" | "name">) {
+type Args = Parameters<typeof readConfig>[1] & { config?: string };
+type Options = Parameters<typeof readConfig>[2];
+
+export function getConfig(args: Args, options?: Options, entryPath?: string) {
 	const configPath =
-		args.config || (args.name && findWranglerConfig(path.dirname(args.name)));
-	return readConfig(configPath, args);
+		args.config || (entryPath && findWranglerConfig(path.dirname(entryPath)));
```

Comment:
In the future I think we could somehow roll this into `readConfig()` itself and remove the need for this helper function at all.